### PR TITLE
Searchable Phone Extension Picker

### PR DESCRIPTION
**Pull request by goergisn**
<changed>
When shoppers select the prefix picker in the phone number input field, they can now search through the prefix list using their ISO code, prefix, or country/region name.
</changed>

----------------

## Summary
- The button that shows the list remains unchanged visually (showing the ISO code, chevron down, prefix)
- The list switches from a "barrel" to a searchable list that allows searching for `ISO code`, `prefix` + `localized country name`
- The list is sorted by localized country name by default (except for `qiwiWallet` and `mbWay`)

### TODO
- [x] Localization (Pending final decisions)

| Before | After |
| ----- | ----- |
| ios | simulator_screenshot_0DCE8ED2-33BB-4481-B6C8-161575888558 |

### CODE REVIEW

File: Adyen/UI/Form/Items/Value Pickers/Phone Extension Picker/FormPhoneExtensionPickerItemView.swift [hunk 1]
```diff
@@ -4,25 +4,78 @@
...
-    override internal func createInputControl() -> PickerTextInputControl {
-        phoneExtensionInputControl
+/// A control to select a phone extension from a list.
+@_spi(AdyenInternal)
+public final class FormPhoneExtensionPickerItemView: FormItemView<FormPhoneExtensionPickerItem> {
+    
+    internal lazy var valueLabel = UILabel(style: item.style.text)
+    
+    internal var accessoryImage: UIImage? { UIImage(
+        named: "chevron_down",
```
Comment (nauaros) [inline Adyen/UI/Form/Items/Value Pickers/Phone Extension Picker/FormPhoneExtensionPickerItemView.swift]: Only suggestion is to move this to a constant? Only if you want, not a big deal

**Review by nauaros — DISMISSED** [review]

File: Adyen/Core/Components/AbstractPersonalInformationComponent/PhoneFormItemInjector.swift [hunk 2]
```diff
@@ -12,29 +12,38 @@ internal final class PhoneFormItemInjector: FormItemInjector, Localizable {
...
     internal let style: FormTextItemStyle
 
-    internal let phoneExtensions: [PhoneExtensionPickerItem]
+    internal let phoneExtensions: [PhoneExtension]
 
     internal var value: PhoneNumber?
 
     internal var identifier: String
+    
+    internal let presenter: WeakReferenceViewControllerPresenter
```
Comment (erenbesel) [inline Adyen/Core/Components/AbstractPersonalInformationComponent/PhoneFormItemInjector.swift:21]: What is this protocol/class? Can't find it on current code

Comment (goergisn) [reply Adyen/Core/Components/AbstractPersonalInformationComponent/PhoneFormItemInjector.swift:21]: It's there for longer already - it's a wrapper that holds a weak reference to a presenter and is itself conforming to the protocol - this way we don't have to make the parameter nil-able

File: Adyen/UI/Form/Items/Value Pickers/Phone Extension Picker/FormPhoneExtensionPickerItemView.swift (hunk 1 above)
Comment (goergisn) [reply Adyen/UI/Form/Items/Value Pickers/Phone Extension Picker/FormPhoneExtensionPickerItemView.swift]: Would make this part of a bigger refactor of moving all `UIImage(named:)` usages to a central place

File: Adyen/UI/Form/FormItemViewBuilder.swift [hunk 3]
```diff
@@ -110,9 +104,15 @@ public struct FormItemViewBuilder {
     
     /// Builds `FormPickerItemView` from `FormPickerItem`.
     @_spi(AdyenInternal)
-    public func build(with item: FormPickerItem) -> FormItemView<FormPickerItem> {
+    public func build<Value>(with item: FormPickerItem<Value>) -> FormItemView<FormPickerItem<Value>> {
```
Comment (nauaros) [inline Adyen/UI/Form/FormItemViewBuilder.swift:107]: Nice!

File: Adyen/UI/Form/Items/Phone Number/Phone Extension Picker/FormPhoneExtensionPickerItemView.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+/// A control to select a phone extension from a list.
+@_spi(AdyenInternal)
+public final class FormPhoneExtensionPickerItemView: FormItemView<FormPhoneExtensionPickerItem> {
+    
+    private lazy var valueLabel = UILabel(style: item.style.text)
+    
+    private lazy var chevronView: UIImageView = {
+        let image = UIImage(
+            named: "chevron_down",
```
Comment (nauaros) [inline Adyen/UI/Form/Items/Phone Number/Phone Extension Picker/FormPhoneExtensionPickerItemView.swift]: What do you think about moving this into a constant?

**Review by nauaros — DISMISSED** [review]
Nice work!

File: Adyen/UI/Form/Items/Phone Number/FormPhoneNumberItem.swift [hunk 5]
```diff
@@ -27,15 +27,19 @@ public final class FormPhoneNumberItem: FormTextItem {
...
-        phonePrefixItem = FormPhoneExtensionPickerItem(
-            preselectedValue: selectableValues.preselectedPhoneNumberPrefix(for: phoneNumber),
-            selectableValues: selectableValues,
-            style: style
+    public init(
+        phoneNumber: PhoneNumber?,
+        selectableValues: [PhoneExtension],
+        style: FormTextItemStyle,
+        localizationParameters: LocalizationParameters? = nil,
+        presenter: ViewControllerPresenter
```
Comment (erenbesel) [inline Adyen/UI/Form/Items/Phone Number/FormPhoneNumberItem.swift]: Shouldn't this be the weak presenter class?

File: Tests/IntegrationTests/Card Tests/Items/CardNumberItem/FormItemViewBuilderTests.swift [hunk 6]
```diff
@@ -10,9 +10,14 @@ import XCTest
...
-        let selectableValues = [PhoneExtensionPickerItem(identifier: "test_id",
-                                                         element: .init(value: "test title", countryCode: "test extension"))]
-        let item = FormPhoneExtensionPickerItem(preselectedValue: selectableValues[0], selectableValues: selectableValues, style: FormTextItemStyle())
+        let selectableValues: [PhoneExtension] = [.init(value: "+31", countryCode: "NL")]
+        let item = FormPhoneExtensionPickerItem(
+            preselectedExtension: selectableValues[0],
+            selectableExtensions: selectableValues,
+            validationFailureMessage: nil,
+            style: .init(),
+            presenter: UIViewController()
```
Comment (erenbesel) [inline Tests/IntegrationTests/Card Tests/Items/CardNumberItem/FormItemViewBuilderTests.swift]: :)

**Review by erenbesel — APPROVED** [review]

**Review by nauaros — APPROVED** [review]

**Pull request merged**